deltametrics/mask.py

```diff
@@ -48,51 +53,308 @@ def show(self, **kwargs):
         """Initialize the ChannelMask.
 
-        Intializing the channel mask requires an array of data, should be
-        two-dimensional.
+        Intializing the channel mask requires a flow velocity field and an
+        array of the delta topography.
 
         Parameters
         ----------
-        arr : ndarray
-            The data array to make the mask from.
+        velocity : ndarray
+            The velocity array to be used for mask creation.
+
+        topo : ndarray
+            The model topography to be used for mask creation.
+
+        Other Parameters
+        ----------------
+        velocity_threshold : float, optional
+            Threshold velocity above which flow is considered 'channelized'.
+            The default value is 0.3 m/s based on DeltaRCM default parameters
+            for sediment transport.
+
+        landmask : :obj:`LandMask`, optional
+            A :obj:`LandMask` object with a defined binary land mask.
+            If given, it will be used to help define the channel mask.
+
+        wetmask : :obj:`WetMask`, optional
+            A :obj:`WetMask` object with a defined binary wet mask.
+            If given, the landmask attribute it contains will be used to
+            determine the channel mask.
 
-        routine : str
-            Which routine to use to extract the mask.
+        topo_threshold : float, optional
+            Threshold depth to use for the OAM. Default is -0.5.
+
+        angle_threshold : int, optional
+            Threshold opening angle used in the OAM. Default is 75 degrees.
+
+        numviews : int, optional
+            Defines the number of times to 'look' for the OAM. Default is 3.
 
         is_mask : bool, optional
             Whether the data in :obj:`arr` is already a binary mask. For
             example, this should be set to True, if you have already binarized
             the data yourself, using custom routines, and want to just store
-            the data in the ChannelMask object.
-        """
-        super().__init__(mask_type='channel', data=arr)
+            the data in the LandMask object.
 
-        self.other_atts = 10
+        """
+        super().__init__(mask_type='channel', data=topo)
+        self.velocity = velocity
+
+        # assign **kwargs to self
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        # set default values for **kwargs if they do not exist
+        self.velocity_threshold = getattr(self, 'velocity_threshold', 0.3)
+        self.topo_threshold = getattr(self, 'topo_threshold', -0.5)
+        self.angle_threshold = getattr(self, 'angle_threshold', 75)
+        self.numviews = getattr(self, 'numviews', 3)
+        self.is_mask = getattr(self, 'is_mask', False)
+
+        if self.is_mask is False:
+            self.compute_channelmask()
+        elif self.is_mask is True:
+            self.mask = self.data
+        else:
+            raise ValueError('is_mask must be a `bool`,'
```

Comment:
did this get hit in tests? I think this will fail if self.is_mask` is not a string already.  

```
>>> a = True
>>> 'the test' + a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: must be str, not bool
```

Could try fixing it with `str(self.is_mask)` I think.

Reply:
Good point, the line was covered in the tests, but the invalid value for `is_mask` was a string 'invalid' so it worked. I hadn't thought about using the `if not` convention but that is a change I can make. And yes I think the `str(self.is_mask)` will help cover cases with numerics or other types.

Reply:
cool. Yeah the safest bet is probably just `type(self.is_mask)`, because I think `str()` will also fail if the object does not have `__repr__` implemented.